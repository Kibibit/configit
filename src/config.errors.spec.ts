import { validateSync } from 'class-validator';
import strip from 'strip-color';

import { ConfigValidationError } from './config.errors';
import { PizzaConfig, ToppingEnum } from './pizza.config.model.mock';

describe('Config Model', () => {
  let validationError: ConfigValidationError;
  beforeEach(() => {
    validationError = new ConfigValidationError([]);
  });
  test('validationError creation', () => {
    expect(validationError).toBeDefined();
  });

  test('ConfigValidationError beautify class-validator errors', () => {
    const config = new PizzaConfig({
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      toppings: [ ToppingEnum.Cheese, ToppingEnum.Pepperoni, 5 as any ]
    });
    const validationErrors = validateSync(config);
    validationError = new ConfigValidationError(validationErrors);

    expect(strip(validationError.message).replace(/(?:\r\n|\r|\n)/g, '\n')).toMatchSnapshot();
  });
});
