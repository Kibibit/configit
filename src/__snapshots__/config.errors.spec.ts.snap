// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Model ConfigValidationError beautify class-validator errors 1`] = `
"
=======================================================
 property: toppings
 value: [ 'cheese', 'pepperoni', 5 ]
=======================================================
    - Topping must be one of: cheese, pepperoni, sausage

"
`;
